checkout-code/action.yml

```diff
@@ -0,0 +1,34 @@
+name: 'Checkout code from PR'
+description: 'Checkout code from PR based either on PR event or comment event in that PR'
+runs:
+  using: "composite"
+  steps:
+  - name: Checkout code from PR event
+    uses: actions/checkout@v2
+    if: ${{ github.event.comment == '' }}
+    with:
+      ref: ${{github.event.pull_request.head.ref}}
+      repository: ${{github.event.pull_request.head.repo.full_name}}
+      fetch-depth: 0
+
+  - name: Send Github API Request to get PR data
+    id: request
+    uses: octokit/request-action@v2.0.0
+    if: ${{ github.event.pull_request == '' }}
+    with:
+      route: ${{ github.event.issue.pull_request.url }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  - name: Print out PR data
+    id: pr_data
+    run: |
+      echo "::set-output name=json::${{ steps.request.outputs.data }}"
+
+  - name: Checkout code from PR
+    uses: actions/checkout@v2
+    if: ${{ github.event.pull_request == '' }}
```

Comment:
I don't completely understand how this step differs from the 'Checkout code from PR event' one but should this only checkout if the comment contains '/retest'?

Reply:
I guess the difference is that this step will work in case of the comment in the PR vs the one based on the PR event.
And the comment content is checked in the downstream code: https://github.com/codeready-toolchain/toolchain-e2e/pull/366/files#diff-2c6a8154e544f9a40f2e6404df6d5ae16d761cd295b2274df73359d89a5002c9R17

Reply:
Yeah, this step will be executed only for events `issue_comment` - the event doesn't contain `github.event.pull_request`. 
Analogically, the previous `Checkout code from PR event` step is executed only for PR events - they don't contain `github.event.comment`
As for the check of the `/retest` comment - this needs to be set at the level of the job itself so the workflow is not even started.